app/controllers/users_controller.rb

```diff
@@ -3,20 +3,21 @@ class UsersController < ApplicationController
   before_action :obtain_user, only: %i[show edit update destroy]
 
   def index
-    @pagy, @users = pagy_decorated(User.all, items: 10)
+    @pagy, @users = pagy_decorated(authorize(User.all), items: 10)
   end
 
   def show; end
 
   def new
-    @user = User.new
+    @user = authorize(current_company.users.build)
   end
 
   def create
-    @user = User.new(user_params)
+    @user = current_company.users.create(user_params)
```

Comment:
```suggestion
    @user = authorize(current_company.build_user(user_params))
```

Потім в моделі Company:
```ruby
def build_user(params)
  role = params.delete(:role)
  user_company_relationships.build(role: role).then do |relationship|
    relationship.build_user(params)
  end
end
```